src/Api/FormatsProvider.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Api;
+
+use ApiPlatform\Core\Exception\InvalidArgumentException;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+
+/**
+ * {@inheritdoc}
+ *
+ * @author Anthony GRASSIOT <antograssiot@free.fr>
+ */
+final class FormatsProvider implements FormatsProviderInterface
+{
+    private $configuredFormats;
+    private $formats;
+    private $resourceMetadataFactory;
+
+    public function __construct(ResourceMetadataFactoryInterface $resourceMetadataFactory, array $formats)
+    {
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->configuredFormats = $formats;
+        $this->formats = [];
+    }
+
+    /**
+     * Finds formats from a Resource class.
+     */
+    public function getFormatsFromAttributes(array $attributes): array
```

Comment:
Not sure if I like this concept here. I mean it's a matter of how we want it to work. Right now the specified formats on the resource would **override** the generally configured formats. But if I think about it, the generally configured formats are **general**. So what I expect this annotation to do is to configure **additional** formats for this specific resource. So imho it should **merge** the configured formats with the ones defined on the resource. I understand that it can be a use case to say everything should be available in `json` **except** for this one resource so maybe how it behaves should be a configuration option too?

Reply:
That's indeed one of the concerns I have too. it could be painful if you add a general format, you'll then need to update all annotation.

Reply:
Yeah for me the general ones are the ones that should be available for **all** endpoints. So merging seems more appropriate to me.

Reply:
I like the idea to override the default formats, it's pretty useful (at the operation level) I think. IMHO, If we merge the configured formats, we also have to find a way to be able to completely override the default formats.

Reply:
Yeah, thinking about `additionalFormats` and `formats`. The first just adds and the other overrides and you're done.

Reply:
ok let's try to get this merged with `formats` only and I'll add `additionalFormats` in a following PR if @api-platform/core-team agrees. That will be pretty easy anyway